### ISSUE TITLE
Products show as out of stock

### ISSUE THREAD

**Issue by dpilarsk** [karliuka/m2.ProductAvailable#12]
Hi,

I have exactly the same problem as https://github.com/karliuka/m2.ProductAvailable/issues/11 or https://github.com/karliuka/m2.ProductAvailable/issues/10
And nothing more in Stores > Configuration > Catalog > Product Available. Nothing mentioned to display 'Out of stock' for guests

Any ideas ?

**Comment by maciejlemiesz** [comment]
I can only confirm this for 2.4.6, for guests it shows "out of stock" Can we somehow change this text?

**Comment by karliuka** [comment]
> I can only confirm this for 2.4.6, for guests it shows "out of stock" Can we somehow change this text?

this text written on your template
![Screenshot_20230731_202351](https://github.com/karliuka/m2.ProductAvailable/assets/1991955/cd530fd0-592f-43d9-a309-0b66eca6cb6e)

**Comment by maciejlemiesz** [comment]
yes, but its general message for information when something is really "Out of stock". I do not want to change it as its useful information for customer if some item is not in stock.
The requirement is to show different label if price is hidden, maybe like "Register to see the price".

**Comment by karliuka** [comment]
for that use can_show_price flag of product in your template

![Screenshot_20230802_115740](https://github.com/karliuka/m2.ProductAvailable/assets/1991955/3e8f9024-5a27-4606-9ab6-7909eb0af862)
result
![Screenshot_20230802_115838](https://github.com/karliuka/m2.ProductAvailable/assets/1991955/bc6f868e-ca3c-4c52-aa38-ad70b5e5785e)

see
https://github.com/karliuka/m2.ProductAvailable/blob/master/Observer/CollectionObserver.php
https://github.com/karliuka/m2.ProductAvailable/blob/master/Observer/ProductObserver.php
